python/nav/web/devicehistory/views.py

```diff
@@ -119,7 +119,7 @@ def devicehistory_view(request, **_):
     if len(set(valid_params) & set(request.GET.keys())) >= 1:
         form = DeviceHistoryViewFilter(request.GET)
     else:
-        form = DeviceHistoryViewFilter()
+        form = DeviceHistoryViewFilter(DeviceHistoryViewFilter.get_initial())
```

Comment:
Normally, it is safer to convert to a QueryDict. A QueryDict only has lists as values and extra methods for fetching the entire list. QueryDict.get() fetches the first element of the lists, QueryDict.get_list() fetches the entire list. This is for multiselect fields no matter if radio-buttons or dropdowns are used.

There are no multiselect fields in DeviceHistoryViewFilter so as long as the tests are good and green don't bother. If there is other weird behavior later the fix might be to go via a QueryDict.

Reply:
I'd be tempted to have a unit test proving that feeding in `get_initial()` and request.GET with the same values as `get_initial()` leads to the same `form.cleaned_data`.

Reply:
That's a good idea! I'm on it!

Reply:
Done, but had to add it to integration tests, since the is magic happening in `DeviceHistoryViewFilter.__init__`